base/src/AbsControlModule.cpp

```diff
@@ -84,11 +92,54 @@ boost::shared_ptr<Module> AbsControlModule::getModuleofRole(std::string role)
 	boost::shared_ptr<Module> moduleWithRole = nullptr;
 	try
 	{
-		moduleWithRole = moduleRoles[role];
+		moduleWithRole = moduleRoles[role].lock();
```

Comment:
Why we need lock here

Reply:
.lock() is the mechanism to get a temporary `shared_ptr` from a `weak_ptr`